pkgs/development/interpreters/python/default.nix

```diff
@@ -14,13 +14,61 @@ with pkgs;
     , packageOverrides
     , sitePackages
     , hasDistutilsCxxPatch
-    , pythonForBuild
-    , self
+    , pythonPackagesBuildBuild
+    , pythonForBuild # provides pythonPackagesBuildHost
+    , pythonPackagesBuildTarget
+    , pythonPackagesHostHost
+    , self # is pythonPackagesHostTarget
+    , pythonPackagesTargetTarget
     }: let
-      pythonPackages = callPackage ../../../top-level/python-packages.nix {
+      pythonPackagesFun = callPackage ../../../top-level/python-packages.nix {
         python = self;
-        overrides = packageOverrides;
       };
+      pythonPackages = lib.makeScopeWithSplicing
+        pkgs.splicePackages
+        pkgs.newScope
+        {
+          selfBuildBuild = pythonPackagesBuildBuild;
+          selfBuildHost = pythonForBuild.pkgs;
+          selfBuildTarget = pythonPackagesBuildTarget;
+          selfHostHost = pythonPackagesHostHost;
+          selfTargetTarget = pythonPackagesTargetTarget;
+        }
+        (self: {
+          inherit (self)
+            isPy27 isPy35 isPy36 isPy37 isPy38 isPy39 isPy3k isPyPy pythonAtLeast pythonOlder
+            python bootstrapped-pip buildPythonPackage buildPythonApplication
+            fetchPypi
+            hasPythonModule requiredPythonModules makePythonPath disabledIf
+            toPythonModule toPythonApplication
+            buildSetupcfg
+
+            eggUnpackHook
+            eggBuildHook
+            eggInstallHook
+            flitBuildHook
+            pipBuildHook
+            pipInstallHook
+            pytestCheckHook
+            pythonCatchConflictsHook
+            pythonImportsCheckHook
+            pythonNamespacesHook
+            pythonRecompileBytecodeHook
+            pythonRemoveBinBytecodeHook
+            pythonRemoveTestsDirHook
+            setuptoolsBuildHook
+            setuptoolsCheckHook
+            venvShellHook
+            wheelUnpackHook
+
+            wrapPython
+
+            pythonPackages
+
+            recursivePthLoader
+          ;
+        })
```

Comment:
we need to extend here with `packageOverrides` to not break compatibility

Reply:
Oh oops, sorry. I used to have that. Will fix.